compiler/aarch64/codegen/ARM64Ops.hpp

```diff
@@ -0,0 +1,1225 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef ARM64OPS_INCL
+#define ARM64OPS_INCL
+
+typedef enum  {
+//		Opcode                                                          	BINARY    	OPCODE    	comments
+/* UNALLOCATED */
+		ARM64Op_bad,                                                    	/* 0x00000000	BAD       	invalid operation */
+/* Branch,exception generation and system Instruction */
+	/* Compare _ Branch (immediate) */
+		ARM64Op_cbz_32_bit,                                             	/* 0x34000000	CBZ       	 */
+		ARM64Op_cbnz_32_bit,                                            	/* 0x35000000	CBNZ      	 */
+		ARM64Op_cbz_64_bit,                                             	/* 0xB4000000	CBZ       	 */
+		ARM64Op_cbnz_64_bit,                                            	/* 0xB5000000	CBNZ      	 */
+	/* Test & branch (immediate) */
+		ARM64Op_tbz,                                                    	/* 0x36000000	TBZ       	 */
+		ARM64Op_tbnz,                                                   	/* 0x37000000	TBNZ      	 */
```

Comment:
That is correct! We can just have different templates for different instructions and ORing them in (this will need to be taught through for vfp, the number of template will get large once included) from memory we would need about 30 ish template for basic instructions (not looking at vfp instructions... This will blow that out of proportion. And we are no where near done with these as a lot of them have vector and scalar variant that calls for different templates ...sigh)).